examples/FCCee/vertex/analysis.py

```diff
@@ -28,42 +28,53 @@ def __init__(self, inputlist, outname, ncpu):
         print (" done")
     #__________________________________________________________
     def run(self):
-        df2 = (self.df.Range(10000)
-        #df2 = (self.df
+        #df2 = (self.df.Range(10000)
+        df2 = (self.df
+
+               # MC event primary vertex
+               .Define("MC_PrimaryVertex",  "getMC_EventPrimaryVertex( Particle )" )
+
                # number of tracks
                .Define("ntracks","get_nTracks(EFlowTrack_1)")
+
                # Select tracks with d0 and z0 significance < 3 sigmas
+		   # note: d0 and z0 are defined w.r.t. (0,0,0)
+		   # hence do not use such criteria to select primary tracks
+		   # if the events were generated with a vertex distribution
                .Define("SelTracks","selTracks(0.,3.,0.,3.)( ReconstructedParticles, EFlowTrack_1)")
                .Define("nSeltracks",  "getRP_n(SelTracks)")
                # Reconstruct the vertex from these tracks :
-               .Define("Vertex",  "VertexFB( 1, SelTracks, EFlowTrack_1 )")	# primary vertex
-               #
+               .Define("Vertex",  "VertexFB( 1, SelTracks, EFlowTrack_1 )")	# primary vertex, in mm
```

Comment:
I don't really like to have to call the vertex fitter VertexFB we should come up with something more clear.